docs/iam_roles.md

```diff
@@ -0,0 +1,103 @@
+# IAM Roles
+
+Two IAM roles are created for the cluster: one for the masters, and one for the nodes.
+The permissions are kept to the minimum required to setup and maintain the cluster.
+
+Master permissions:
+
+```
+ec2:*
+route53:*
+elasticloadbalancing:*
+ecr:GetAuthorizationToken
+ecr:BatchCheckLayerAvailability
+ecr:GetDownloadUrlForLayer
+ecr:GetRepositoryPolicy
+ecr:DescribeRepositories
+ecr:ListImages
+ecr:BatchGetImage
+
+// The following permissions are only created if you are using etcd volumes with "encrypted: true" and a custom kmsKeyId.
+// They are scoped to the kmsKeyId that you are using.
+kms:Encrypt
+kms:Decrypt
+kms:ReEncrypt*
+kms:GenerateDataKey*
+kms:DescribeKey
+kms:CreateGrant
+kms:ListGrants
+kms:RevokeGrant
+```
+
+Node permissions:
+
+```
+ec2:Describe*
+route53:*
+ecr:GetAuthorizationToken
+ecr:BatchCheckLayerAvailability
+ecr:GetDownloadUrlForLayer
+ecr:GetRepositoryPolicy
+ecr:DescribeRepositories
+ecr:ListImages
+ecr:BatchGetImage
+```
+
+## Adding Additional Permissions
+
+Sometimes the default permissions may not be enough for your needs, and you will have to add more. You can do this
+through the `additionalMasterPermissions` and `additionalNodePermissions` spec fields. For instance, let's say you want
+to add DynamoDB and Elasicsearch permissions to your nodes.
+
+Edit your cluster via `kops edit cluster ${CLUSTER_NAME}` and add the following to the spec:
```

Comment:
A lot of kops features require the three-step process of create-edit-update. I don't see how this feature can be easily converted into a flag on `kops create`. Besides, no external resources are actually being created until the final update step, so at no point does the cluster exist without the extra IAM permissions.